src/Features/CSharp/Portable/Completion/KeywordRecommenders/UsingKeywordRecommender.cs

```diff
@@ -126,14 +118,62 @@ private static bool IsUsingDirectiveContext(CSharpSyntaxContext context, Cancell
             // |
             if (token.Kind() == SyntaxKind.SemicolonToken)
             {
-                if (token.Parent.IsKind(SyntaxKind.ExternAliasDirective) ||
-                    token.Parent.IsKind(SyntaxKind.UsingDirective))
+                if (token.Parent.IsKind(SyntaxKind.ExternAliasDirective, SyntaxKind.UsingDirective))
                 {
                     return true;
                 }
             }
 
+            // extern alias a;
+            // global |
+
+            // global using Goo;
+            // global |
+
+            // global |
+            if (!forGlobalKeyword)
+            {
+                var previousToken = token.GetPreviousToken(includeSkipped: true);
+                if (previousToken.Kind() == SyntaxKind.None)
+                {
+                    // root namespace
+                    if (token.Kind() == SyntaxKind.GlobalKeyword)
+                    {
+                        return true;
+                    }
+                    else if (token.Kind() == SyntaxKind.IdentifierToken && SyntaxFacts.GetContextualKeywordKind((string)token.Value!) == SyntaxKind.GlobalKeyword)
+                    {
+                        return IsValidContextAtTheRoot(context, originalToken, cancellationToken);
+                    }
+                }
+                else if (previousToken.Kind() == SyntaxKind.SemicolonToken &&
+                    previousToken.Parent.IsKind(SyntaxKind.ExternAliasDirective, SyntaxKind.UsingDirective))
+                {
+                    if (token.Kind() == SyntaxKind.GlobalKeyword)
+                    {
+                        return true;
+                    }
+                    else if (token.Kind() == SyntaxKind.IdentifierToken && SyntaxFacts.GetContextualKeywordKind((string)token.Value!) == SyntaxKind.GlobalKeyword)
+                    {
+                        return true;
+                    }
```

Comment:
so there's a *slight* corner here that you could fixup if you wanted.  it looks like with this that if you have:

```
extern alias Foo;
```

tehn we won't offer here:

```
$$
extern alias Foo;
```

but if you have:

```
extern alias Foo;
$$
extern alias Bar;
```

taht we will offer.

this is likely so minor as to be irrelevant though :)

Reply:
> so there's a slight corner here that you could fixup if you wanted.  it looks like with this that if you have:

I'd say that this is a preexisting condition and the current behavior around offering ```using``` keyword (when `global`) isn't present. I'll open an issue to decide if we want that addressed. 

---
In reply to: [600021445](https://github.com/dotnet/roslyn/pull/52059#discussion_r600021445) [](ancestors = 600021445)